### PR TITLE
Refactor/cleanup jns

### PR DESCRIPTION
**Pull request by jduffy3**

### CODE REVIEW

File: src/helpers/RangeFinder.js [hunk 1]
```diff
@@ -7,6 +7,17 @@ function search(element, phrase) {
   }));
 }
 
-export function findRanges(node, patternsToFind) {
-  return patternsToFind.flatMap((pattern) => search(node, pattern));
+export function findRanges(target, patternsToFind) {
```
Comment (sbrudz) [inline src/helpers/RangeFinder.js:10]: `findRanges` doesn't seems like an accurate name.  I know it's been in the code for quite a while, but your refactorings have helped me realize that it's not just finding ranges.  It's returning a list of all the warnings for a particular email plus metadata like range.  This function contains the core business logic of Just Not Sorry.  We need a name that reflects its purpose.  Maybe `calculateWarnings`?

**Review by sbrudz — COMMENTED** [review]
These refactorings improve the clarity of the code.  Great work!

Also, performance on long emails with many warnings is much better now.

While testing, I noticed that the tooltip has a lower z index than the subject line:
image

**Comment by jduffy3** [comment]
> These refactorings improve the clarity of the code. Great work!
> 
> Also, performance on long emails with many warnings is much better now.
> 
> While testing, I noticed that the tooltip has a lower z index than the subject line: image

I wasn't able to reproduce unless I hardcoded the position, but I tried removing the setting of that variable and I think the placement is better. Could you try on your machine and see if that fixes it?

**Comment by sbrudz** [comment]
> I wasn't able to reproduce unless I hardcoded the position, but I tried removing the setting of that variable and I think the placement is better. Could you try on your machine and see if that fixes it?

I pulled down the new code and restarted my local environment.  Now I can't reproduce it anymore.  The warnings always display below the underline when close to the top.  Let's call this issue fixed.

**Review by sbrudz — APPROVED** [review]
Looks good.  Thank you!

**Pull request merged**